lib/AST/Type.cpp

```diff
@@ -2316,6 +2316,9 @@ class IsBindableVisitor
       llvm::DenseMap<std::pair<CanType, ProtocolDecl*>, ProtocolConformanceRef> addedConformances;
       
       for (auto proto : upperBound->getConformsTo()) {
+        if (proto->getInvertibleProtocolKind())
```

Comment:
For classes this is going to produce a builtin conformance which has a decl context anchored on the protocol that is complete different from context of the `genericSig`. Maybe it's better to filter based `isa<BuiltinProtocolConformance>(nomConformance)`.